pandas/core/strings/accessor.py

```diff
@@ -209,8 +212,12 @@ def _validate(data):
         # see _libs/lib.pyx for list of inferred types
         allowed_types = ["string", "empty", "bytes", "mixed", "mixed-integer"]
 
-        values = getattr(data, "values", data)  # Series / Index
-        values = getattr(values, "categories", values)  # categorical / normal
+        # TODO: avoid kludge for tests.extension.test_numpy
```

Comment:
umm cn you avoid this?

Reply:
can revert to use the `getattr` pattern, either way is an anti-pattern